src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AppointmentsService.kt

```diff
@@ -82,27 +86,32 @@ class AppointmentsService(
     this.date!! <= other.date && this.endTime!! <= other.startTime
 
   fun createAppointmentForProbation(videoBooking: VideoBooking, prisoner: PrisonerDetails) {
-    with(prisoner.appointments.single()) {
-      require(type!!.isProbation) {
-        "Appointment type $type is not valid for probation appointments"
-      }
+    checkProbationAppointments(prisoner.appointments, prisoner.prisonCode!!)
 
-      checkExistingProbationAppointmentDateAndTimesDoNotOverlap(prisoner.prisonCode!!)
-      locationValidator.validatePrisonLocation(prisoner.prisonCode, this.locationKey!!)
-
-      // Add the appointment to the booking - it will be saved when the booking is saved
+    with(prisoner.appointments.single()) {
       videoBooking.addAppointment(
         prisonCode = prisoner.prisonCode,
         prisonerNumber = prisoner.prisonerNumber!!,
-        appointmentType = this.type.name,
+        appointmentType = this.type!!.name,
         date = this.date!!,
         startTime = this.startTime!!,
         endTime = this.endTime!!,
-        locationKey = this.locationKey,
+        locationKey = this.locationKey!!,
       )
     }
   }
 
+  fun checkProbationAppointments(appointments: List<Appointment>, prisonCode: String) {
```

Comment:
I can see what you are doing but not sure I fully agree with this.

It would/should never get this far if there is more than one appointment, there should only be a single appointment for a probation request anyway?

Reply:
But this is now public, was it public before?